src/OSPSuite.Core/Services/ResidualsVsTimeChartService.cs

```diff
@@ -0,0 +1,93 @@
+using System;
+using OSPSuite.Core.Domain;
+using System.Collections.Generic;
+using OSPSuite.Core.Chart;
+using OSPSuite.Core.Domain.Data;
+using OSPSuite.Core.Domain.ParameterIdentifications;
+using OSPSuite.Core.Domain.UnitSystem;
+using OSPSuite.Core.Extensions;
+
+namespace OSPSuite.Core.Services
+{
+   public interface IResidualsVsTimeChartService
+   {
+      /// <summary>
+      ///    This will be the name of the curve added to the chart representing the Zero Curve
+      /// </summary>
+      string Zero { get; }
+
+      /// <summary>
+      ///    Adds the zero marker curve to the <paramref name="chart" /> where the residuals equal zero. The zero line
+      /// will begin at <paramref name="minObservedDataTime" /> and end at <paramref name="maxObservedDataTime" />
+      /// </summary>
+      DataRepository AddZeroMarkerCurveToChart(AnalysisChartWithLocalRepositories chart, float minObservedDataTime, float maxObservedDataTime);
+
+      /// <summary>
+      ///    Creates the DataRepository for the given residuals <paramref name="outputResidual" /> with given
+      /// <paramref name="id" /> and <paramref name="repositoryName" /> 
+      /// </summary>
+      DataRepository CreateScatterDataRepository(string id, string repositoryName, OutputResiduals outputResidual);
+   }
+
+   public class ResidualsVsTimeChartService : IResidualsVsTimeChartService
```

Comment:
Let's add a test for this service. I feel that this can be completely tested

Reply:
@msevestre working on this. Also there are a bunch of broken tests that I had not noticed ( no build on the PR, since it gets merged into a feature branch )

Reply:
@msevestre the tests for the service have been added. This should be ready for merge.